New tests

tagirshin opened the pull request:
I've seen that `chython` does not work on the Mac mini M4, and it seems that the [py-mini-racer](https://github.com/sqreen/PyMiniRacer) repo is no longer supported. There is a new fork called [mini-racer](https://github.com/bpcreech/PyMiniRacer) which adds the needed functionality and seems to be the official place now. So I've updated the requirements and also decided to add more tests to check if something fails. I will add more tests in the next days. 

@stsouko could I ask you to check them if you have time and also confirm if `chython` can be migrated to Python 3.10 (required by higher version of Numpy)?

tagirshin committed:
added more tests + fix mini-racer to run on mac ARM

tagirshin committed:
added tests for thiele

stsouko commented on chython/algorithms/calculate2d/__init__.py:
keep it as is for 3.8 compatibility. still required.

stsouko commented on pyproject.toml:
```diff
@@ -30,14 +30,14 @@ include = [
 ]
 
 [tool.poetry.dependencies]
-python = '^3.8'
```
still required

stsouko commented on pyproject.toml:
```diff
@@ -30,14 +30,14 @@ include = [
+mini-racer = '^0.12.4'
 chytorch-rxnmap = {version = '^1.4', optional = true}
 rdkit = {version = '^2023.9', optional = true}
+numpy = "^2.2.2"
```
1.x is still in use unfortunately

stsouko commented on tests/algorithms/aromatics/test_kekule.py:
lets move tests into main codebase as test subpackages

tagirshin replied:
Just a question, we should not import these tests from Chython, right? What I've seen from other packages and devs, they usually just keep the same structure of the package in a separate test directory. But I'm ok with yours solution too, just wondering about the rationale

stsouko replied:
https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-as-part-of-application-code

Both ways are good. I just picked this one

tagirshin replied:
I see, will redo this, thanks for the answer!

stsouko commented on tests/algorithms/aromatics/test_kekule.py:
```diff
@@ -0,0 +1,154 @@
+    # Verify alternating single and double bonds
+    bonds = mol._bonds
+    double_bonds = sum(1 for n, ms in bonds.items() for m, b in ms.items() if b.order == 2 and m > n)
+    assert double_bonds == 3  # benzene should have 3 double bonds
```
easier to assert mol == smiles('C1=CC=CC=C1')

stsouko commented on tests/algorithms/aromatics/test_kekule.py:
```diff
@@ -0,0 +1,154 @@
+    # Test pyridine and its derivatives
+    mol = smiles('n1ccccc1')  # pyridine
+    assert mol.kekule()
+    
```
good to have expected kekule form assertion

stsouko commented on tests/algorithms/aromatics/test_kekule.py:
```diff
@@ -0,0 +1,154 @@
+    # Test pyrrole and its derivatives
+    mol = smiles('[nH]1cccc1')  # pyrrole
+    assert mol.kekule()
+    
```
test for implicit H fixup: smiles('n1cccc1')  smiles('N1C=CC=C1')

stsouko commented on tests/algorithms/fingerprints/test_linear.py:
```diff
@@ -0,0 +1,156 @@
+def test_linear_fingerprint_consistency():
+    # Test that fingerprints are consistent for the same molecule
+    mol = smiles('CCO')
+    fp1 = mol.linear_fingerprint()
```
fp1 = smiles('CCO').linear_fingerprint()
fp2 = smiles('OCC').linear_fingerprint()
to test permutation stability too

stsouko commented on tests/algorithms/fingerprints/test_linear.py:
```diff
@@ -0,0 +1,156 @@
+    # Test different radius parameters
+    fp1 = mol.linear_fingerprint(min_radius=1, max_radius=2)
+    fp2 = mol.linear_fingerprint(min_radius=1, max_radius=3)
+    assert fp2.sum() >= fp1.sum()  # More radius should capture more features
```
lets change to smth like (fp1 & fp2) == fp1 (it's pseudocode!, not correct np impleentation)

stsouko commented on tests/algorithms/fingerprints/test_linear.py:
```diff
@@ -0,0 +1,156 @@
+    fp3 = mol.linear_fingerprint(number_bit_pairs=4)
+    
+    # More bit pairs should potentially activate more bits
+    assert fp1.sum() <= fp2.sum() <= fp3.sum()
```
nbp=1 is subset of nbp=2, etc. so, better to test this

tagirshin committed:
Add GitHub Actions workflow and update tests

- Add GitHub Actions workflow for running tests on Python 3.8-3.11
- Add comprehensive tests for aromatic systems in test_thiele.py
- Update project dependencies in pyproject.toml

tagirshin commented on tests/algorithms/calculate2d/test_clean2d.py:
```diff
@@ -0,0 +1,163 @@
+import pytest
+
+
+def test_molecule_clean2d():
```
Okay, I will remove the tests

tagirshin committed:
test github actions

tagirshin committed:
ci: add GitHub Actions workflow for testing

tagirshin committed:
ci: improve GitHub Actions workflow stability

- Add explicit Cython installation
- Increase timeout for dependency installation
- Add verbose output for debugging
- Configure Poetry worker count

tagirshin committed:
ci: install dev dependencies in GitHub Actions

- Add --with dev flag to poetry install

tagirshin committed:
fix: return compatability with python 3.8

tagirshin committed:
ci: removed python 3.8 for tests

tagirshin committed:
ci: removed some non-needed steps

tagirshin committed:
ci: test approach on compilation

tagirshin committed:
ci: added windows for testing

tagirshin committed:
ci: make windows tests great again

tagirshin committed:
ci: fix build dependencies for Linux and Windows

tagirshin committed:
ci: add pytest plugins for HTML reports and coverage

tagirshin commented:
Oh, so you did merge actually... 
I wanted to fix GitHub actions, so I will create a new MR if you don't mind

stsouko commented:
Check branch V2.
I reactor some tests already
